src/main/java/slimeknights/tconstruct/smeltery/tileentity/CastingBasinTileEntity.java

```diff
@@ -0,0 +1,18 @@
+package slimeknights.tconstruct.smeltery.tileentity;
+
+import net.minecraft.item.crafting.IRecipeType;
+import net.minecraft.tileentity.TileEntityType;
+import slimeknights.tconstruct.library.recipe.RecipeTypes;
+import slimeknights.tconstruct.smeltery.TinkerSmeltery;
+import slimeknights.tconstruct.smeltery.recipe.AbstractCastingRecipe;
+
+public class CastingBasinTileEntity extends AbstractCastingTileEntity {
+
+  public CastingBasinTileEntity() {
+    this(TinkerSmeltery.basin.get(), RecipeTypes.CASTING_BASIN);
+  }
+
+  public CastingBasinTileEntity(TileEntityType<?> tileEntityTypeIn, IRecipeType<AbstractCastingRecipe> recipeType) {
```

Comment:
Remove this constructor, if you are going to call with existing types you should call the abstract.